src/SlimBootstrap/Authentication/Jwt.php

```diff
@@ -65,18 +77,19 @@ public function __construct(
     public function authenticate(Message\ServerRequestInterface $request): array
     {
         try {
-            $publicKey = $this->getPublicKey();
-            $token     = $this->getToken($request);
+            $token = $this->getToken($request);
 
-            $this->verifyToken($token, $publicKey);
-            $this->validateToken($token);
+            $this->jwtConfig->validator()->assert(
+                $token,
+                ...$this->jwtConfig->validationConstraints()
+            );
 
             return [
-                'clientId' => $token->getClaim($this->clientDataClaims['clientId']),
-                'role'     => $token->getClaim($this->clientDataClaims['role']),
+                'clientId' => $token->claims()->get($this->clientDataClaims['clientId']),
+                'role'     => $token->claims()->get($this->clientDataClaims['role']),
             ];
         } catch (\Throwable $exception) {
-            $this->logger->addInfo($exception->getMessage());
+            $this->logger->addWarning($exception->getMessage());
```

Comment:
Are we sure that we want to change the severity?

Reply:
I would say yes. It should be more visible for us when someones use an invalid token.